functions/src/import-geojson.ts

```diff
@@ -81,25 +81,64 @@ export function importGeoJsonCallback(
     );
     // Pipe file through JSON parser lib, inserting each row in the db as it is
     // received.
-    let geoJsonType: any = null;
-    file.pipe(JSONStream.parse('type', undefined)).on('data', (data: any) => {
-      geoJsonType = data;
-    });
+    file
+      .pipe(JSONStream.parse('type', undefined))
+      .on('data', (geoJsonType: string | undefined) => {
+        if (!geoJsonType) {
+          return error(
+            HttpStatus.BAD_REQUEST,
+            'Invalid GeoJSON: Missing "type" property'
+          );
+        }
+        if (geoJsonType !== 'FeatureCollection') {
+          return error(
+            HttpStatus.BAD_REQUEST,
+            `Unsupported GeoJSON Type: Expected 'FeatureCollection', got '${geoJsonType}'`
+          );
+        }
+      });
 
     file
-      .pipe(JSONStream.parse(['features', true], undefined))
-      .on('data', (geoJsonLoi: any) => {
-        try {
-          if (geoJsonType !== 'FeatureCollection') {
+      .pipe(JSONStream.parse('crs', undefined))
+      .on(
+        'data',
+        (
```

Comment:
@rfontanarosa Now might be a good time to move this handler into its own function (`onGeoJsonCrs()`?)